### PR TITLE
New Invitations API Coming in BuddyPress 5.0

### LINKED ISSUE #517: Repair function for bp_groups_get_group_roles() in BB Platform
**Describe the bug**
This BuddyBoss Platform API plugin is unable to run at all, it just generates 500 errors internally when trying to fetch APIs.

The reason is due to this function which does not yet exist in BuddyBoss Platform:
bp_groups_get_group_roles() 

That function is related to a deeper issue:
https://codex.buddypress.org/releases/version-5-0-0/
https://bpdevel.wordpress.com/2019/09/16/new-invitations-api-coming-in-buddypress-5-0/

In BuddyPress 5.0 they edited code related to Group Memberships and Invites, to use the API. Platform does not have this code. The API is expecting those updated functions and failing.

**To Reproduce**
1. Activate BuddyBoss Platform API plugin.
2. Activate BuddyBoss Platform.
3. Add this to your site URL /wp-json/buddypress/v1/friends
4. Turn on wp_debug in config file
4. See errors

**Expected behavior**
The functions need to exist in BuddyBoss Platform and work. And then the API from our needs to output correct data for the API. You can try our plugin with BuddyPress instead and you will see the correct API outputs. This will need to be repaired in a BuddyBoss Platform branch, and at the same time, you will need to revert Ketan's commits in API plugin from Jan 23, 2020 - https://prnt.sc/qs0t3a

**Video**
Video Explanation - https://www.loom.com/share/661525dc81c448c4936626e410ca6d9d

**Related Issues**
https://github.com/buddyboss/buddyboss-platform-api/issues/2

### PR DESCRIPTION
**Pull request by KartikSuthar**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:
In BuddyPress 5.0 they edited code related to Group Memberships and Invites, to use the API. Platform does not have this code. The API is expecting those updated functions and failing.

https://github.com/buddyboss/buddyboss-platform-api/issues/2

Fixes #517.

### How to test the changes in this Pull Request:

1. Check the group invites works.
2. Check the membership request within group work.
3. Check emails with all the invitations functionality with groups.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: src/bp-core/classes/class-bp-email-tokens.php [hunk 1]
```diff
@@ -1130,7 +1130,7 @@ public function token__group_invite_message( $bp_email, $formatted_tokens, $toke
 			return '';
 		}
 
-		if ( empty( $formatted_tokens['invites.message'] ) ) {
+		if ( empty( $tokens['invite.message'] ) ) {
```
Comment (chetansatasiya) [inline src/bp-core/classes/class-bp-email-tokens.php]: @KartikSuthar why we changed the `$formatted_tokens['invites.message']` to `$tokens['invite.message'] ` ? Can you explain me more?

File: src/bp-core/classes/class-bp-email-tokens.php [hunk 2]
```diff
@@ -1152,7 +1152,7 @@ public function token__group_invite_message( $bp_email, $formatted_tokens, $toke
 						<tr>
 							<td>
 								<div style="color: <?php echo esc_attr( $settings['body_text_color'] ); ?>; font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; font-size: <?php echo esc_attr( $settings['body_text_size'] . 'px' ); ?>; letter-spacing: -0.24px; line-height: <?php echo esc_attr( floor( $settings['body_text_size'] * 1.625 ) . 'px' ); ?>;">
-									<?php echo wpautop( $formatted_tokens['invites.message'] ); ?>
+									<?php echo wpautop( $tokens['invite.message'] ); ?>
```
Comment (chetansatasiya) [inline src/bp-core/classes/class-bp-email-tokens.php]: @KartikSuthar why we changed the $formatted_tokens['invites.message'] to $tokens['invite.message'] ? Can you explain me more?

**Comment by chetansatasiya** [comment]
@KartikSuthar Also I am not getting any email from group invitations. @anoopmishrabuddyboss Need you to test this branch and make sure email should be fired. @anoopmishrabuddyboss If you need more help ask @KartikSuthar

File: src/bp-core/classes/class-bp-email-tokens.php (hunk 1 above)
Comment (KartikSuthar) [reply src/bp-core/classes/class-bp-email-tokens.php]: @chetansatasiya, due to parameter change.
https://github.com/buddyboss/buddyboss-platform/pull/586/files#diff-d1844393616e1cb51618374627a1ff71R368

File: src/bp-core/classes/class-bp-email-tokens.php (hunk 2 above)
Comment (KartikSuthar) [reply src/bp-core/classes/class-bp-email-tokens.php]: @chetansatasiya, due to parameter change.
https://github.com/buddyboss/buddyboss-platform/pull/586/files#diff-d1844393616e1cb51618374627a1ff71R368

**Comment by anoopmishrabuddyboss** [comment]
Tested it on my local with the latest `api/invitations` branch working fine. as email is not fired on my local. email functionality is tested with @KartikSuthar and on @KartikSuthar local email, functionality is working fine.
@chetansatasiya please review and merge to the dev branch.

**Comment by chetansatasiya** [comment]
@anoopmishrabuddyboss @KartikSuthar there is an error coming when I create a group and invite tab to see here https://prnt.sc/r7qat3 and group admin can't invite anyone.

@anoopmishrabuddyboss I am curious how it's working for you it should not be worked in your local as well. Can you make sure and give your thoughts on this?

**Comment by anoopmishrabuddyboss** [comment]
@chetansatasiya it's working at my local on with `api/invitations` branch please review once and let me know if missing anything.  
Video:- https://www.loom.com/share/32ea5a81830545b08641522933382537
Screenshots:- https://prnt.sc/r7qmf1 https://prnt.sc/r7qmzw

**Pull request merged**